File: app/src/main/java/com/example/tesonet/database/daos/TokenDao.java
package com.example.tesonet.database.daos;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.example.tesonet.database.models.Token;

@Dao
public interface TokenDao {

    @Insert
    void insert(Token token);

    @Update
    void update(Token token);

    @Query("DELETE FROM token_table")
    void deleteToken();

    @Query("SELECT * FROM token_table")
    LiveData<Token> getTokenValue();
}

File: app/src/main/java/com/example/android_party/data/TokenDao.kt
package com.example.android_party.data

import androidx.room.*

@Dao
interface TokenDao {
    @Query("SELECT * FROM Token")
    fun get(): Token

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(token: Token)

    @Query("DELETE FROM Token")
    fun delete()
}